src/mgm.rs

```diff
@@ -242,21 +248,10 @@ impl MgmKey {
     ///
     /// Returns an error if the Yubikey's default algorithm is unsupported.
     pub fn get_default(yubikey: &YubiKey) -> Result<Self> {
-        match yubikey.version() {
-            // Initial firmware versions default to 3DES.
-            Version { major: ..=4, .. }
-            | Version {
-                major: 5,
-                minor: ..=6,
-                ..
-            } => Ok(Self(MgmKeyKind::Tdes(DEFAULT_MGM_KEY.into()))),
-            // Firmware 5.7.0 and above default to AES-192.
-            Version {
-                major: 5,
-                minor: 7..,
-                ..
-            }
-            | Version { major: 6.., .. } => Ok(Self(MgmKeyKind::Aes192(DEFAULT_MGM_KEY.into()))),
+        match MgmAlgorithmId::default_for_version(yubikey.version()) {
+            MgmAlgorithmId::ThreeDes => Ok(Self(MgmKeyKind::Tdes(DEFAULT_MGM_KEY.into()))),
+            MgmAlgorithmId::Aes192 => Ok(Self(MgmKeyKind::Aes192(DEFAULT_MGM_KEY.into()))),
+            _ => Err(Error::NotSupported),
```

Comment:
This is the only part that's different, AFAICT, but also it's currently unreachable